libr/asm/arch/arm/armass.c

```diff
@@ -456,17 +456,15 @@ static ut64 getnum(const char *str) {
 static ut64 getnumbang(const char *str) {
 	ut64 res;
 
-	if ((strlen (str) < 1) || (str[strlen (str) - 1] != '!')) {
+	if (!str || (strlen (str) < 1) || (str[strlen (str) - 1] != '!')) {
```

Comment:
If strlen==0 just do !*str, no need to walk the whole string to determine if its empty

Reply:
For readability use r_Str_endswith() instead of the strlen-1 thing